qiskit/providers/basicaer/qasm_simulator.py

```diff
@@ -387,18 +385,11 @@ def _validate_measure_sampling(self, experiment):
             # measure sampling is allowed
             self._sample_measure = True
 
-    @deprecate_arg(
-        "qobj",
-        deprecation_description="Using a qobj for the first argument to QasmSimulatorPy.run()",
-        since="0.22.0",
-        pending=True,
-        predicate=lambda qobj: not isinstance(qobj, (QuantumCircuit, list)),
-    )
-    def run(self, qobj, **backend_options):
-        """Run qobj asynchronously.
+    def run(self, qcirc, **backend_options):
+        """Run qcirc asynchronously.
 
         Args:
-            qobj (Qobj): payload of the experiment
+            qcirc (Union[QuantumCircuit, List[QuantumCircuit]]): circuit of the experiment
             backend_options (dict): backend options
```

Comment:
From what I have understood, this deprecation was introduced when the interface of `BasicAer` backends (for example, `QasmSimulatorPy`) was updated to follow the interface of `BackendV1`. If you look at the `BackendV1` class [(link)](https://github.com/Qiskit/qiskit/blob/main/qiskit/providers/backend.py#L213), the `run` method has a `run_input` field where you input the circuit, so it would probably make sense to use the name `run_input` instead of `qcirc` here.

Reply:
Thank you so much for pointing it out. I will make the necessary edits and push the commit for review.